### PR TITLE
#158320071 User can filter by warranty 

### PR DESCRIPTION
**Pull request by Dammyy**
**What does this PR do?**

Implements feature that allows a user to filter assets in a single category by warranty

**Description of Task to be completed?**

filter assets in a single category by warranty using the value of two specified dates

**How should this be manually tested?**

click on any asset category card from the dashboard
click the filter dropdown
select two dates in the warranty date input section of the filter dropdown

**Any background context you want to provide?**

none

**What are the relevant pivotal tracker stories?**

[#158320071](https://www.pivotaltracker.com/story/show/158320071)

**Screenshots (if appropriate)**

screen shot 2018-07-26 at 10 16 08 am
screen shot 2018-07-26 at 10 16 27 am

**Questions:**

none

### CODE REVIEW

File: src/pages/SingleCategoryPage/index.tsx [hunk 1]
```diff
@@ -84,6 +93,45 @@ export class SingleCategoryPage extends React.Component<SingleCategoryPageProps,
...
+   *
+   * @param {String} assetCategoryId
+   * @param {String} searchTerm
+   * @param {DateAddedField} dateAdded
+   */
+  searchAssets = (
+    assetCategoryId = this.props.match.params.assetCategoryId,
+    searchTerm = this.state.searchTerm,
+    dateAdded = this.state.dateAdded,
+    warranty = this.state.warranty
```
Comment (kevinejiro) [inline src/pages/SingleCategoryPage/index.tsx]: Shouldn't `warranty` be in the documentation too ?

Comment (Dammyy) [reply src/pages/SingleCategoryPage/index.tsx]: Added. Thanks for pointing it out.

**Comment by Dammyy** [comment]
@andela-kakpobome I have fixed the issues.

**Comment by andela-kakpobome** [comment]
Please fix conflicts here

**Comment by Dammyy** [comment]
@andela-kakpobome  Merge conflicts have been fixed.

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 2]
```diff
@@ -130,38 +140,43 @@ describe('SingleCategoryPage component', () => {
...
   });
 
+  it('clears searchTerm state if clearFilter is clicked when searchTerm < minSearchCriteria', () => {
+    const searchTerm = 'ol';
+    const wrapper = mount(<MemoryRouter><SingleCategoryPage { ...props } /></MemoryRouter>);
+    wrapper.find('.filter-button').simulate('click');
+    const searchInputElement = wrapper.find('.search-input');
+    const singleCategoryPageComp = wrapper.find(SingleCategoryPage);
+
+    searchInputElement.simulate('change', { target: { value: searchTerm } });
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: You still did not check that the state was this value before checking that it was empty after clicking the clear filter

**Pull request merged**